src/features/posts/posts.ts

```diff
@@ -0,0 +1,54 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type InitialState = {
+  loaded: boolean;
+  hasError: boolean;
+  posts: Post[];
+  selectedPost: Post | null;
+};
+
+const initialState: InitialState = {
+  loaded: false,
+  hasError: false,
+  posts: [],
+  selectedPost: null,
+};
+
+export const init = createAsyncThunk(
+  'posts/fetch',
+  (userId: number | object) => {
```

Comment:
The `init` thunk accepts a `userId` parameter that can be a number or an object. This might lead to unexpected behavior. Consider restricting the parameter type to `number` only to ensure consistent behavior.